### ISSUE TITLE
Subtitle Editor: It should not be possible to have overlapping subtitles

### ISSUE THREAD

**Issue by snoesberger** [opencast/opencast-editor#1396]
We don't think it should be possible to have overlapping subtitles. For example, if a user drags a subtitle box in the subtitle editor timeline and moves it over an existing box, the dragging box should shrink so that it starts/ends directly after/before the other box. The box can shrink to a minimal length and then can't be moved any further.

**Comment by Arnei** [comment]
> We don't think it should be possible to have overlapping subtitles.

Disagree with this. The WebVTT standard allows for overlapping cues. And we should accept any WebVTT as long as it is valid WebVTT. I am not in favor of disallowing or breaking perfectly valid subtitles.

I understand that subtitle editor usability could be better. But, I am also not generally in favor of preventing cues from overlapping while editing, as it is valid and likely necessary for some users.  As such, I would ask that discussions on how to improve usability keep this in mind.

**Comment by snoesberger** [comment]
Ok, I see your point. If someone uploads a valid vtt file with overlapping subtitles, we could run into all sorts of problems disallowing overlapping.

**Comment by mtneug** [comment]
> > We don't think it should be possible to have overlapping subtitles.
> 
> Disagree with this. The WebVTT standard allows for overlapping cues. And we should accept any WebVTT as long as it is valid WebVTT. I am not in favor of disallowing or breaking perfectly valid subtitles.
> 
> I understand that subtitle editor usability could be better. But, I am also not generally in favor of preventing cues from overlapping while editing, as it is valid and likely necessary for some users. As such, I would ask that discussions on how to improve usability keep this in mind.

Yes, please follow the standard and allow all valid WebVTT files.
